src/rez/utils/filesystem.py

```
@@ -690,3 +692,56 @@ def windows_long_path(dos_path):
         path = "\\\\?\\" + path
 
     return path
+
+
+def rename(src, dst):
+    """Utility functions to rename a file or folder src to dst with retrying.
+
+    This function uses the built-in 'os.rename()' method, but in case of a
+    'PermissionError', it uses the 'robocopy' command line utility on Windows.
+    This is done to avoid the frequent 'Access is denied' exception raised by 'os.rename()' on Windows.
+
+    Args:
+        src (str): The original name (path) of the file or folder.
+        dst (str): The new name (path) for the file or folder.
+
+    Raises:
+        OSError: If renaming fails after all attempts.
+
+    References:
+        - https://github.com/conan-io/conan/blob/develop2/conan/tools/files/files.py#L207
+
+    """
+    try:
+        os.rename(src, dst)
+    except PermissionError:
+        if is_windows and which("robocopy") and os.path.isdir(src):
```

The exception will be eaten by the except block if this condition is not true.